test/nexus_test.go

```diff
@@ -517,6 +555,72 @@ func TestAsyncOperationFromWorkflow(t *testing.T) {
 		}, callerWorkflow, "succeed")
 		require.NoError(t, err)
 		require.NoError(t, run.Get(ctx, nil))
+
+		// Check the link is added in the caller workflow.
+		iter := tc.client.GetWorkflowHistory(
+			ctx,
+			run.GetID(),
+			run.GetRunID(),
+			false,
+			enums.HISTORY_EVENT_FILTER_TYPE_ALL_EVENT,
+		)
+		var targetEvent *historypb.HistoryEvent
+		for iter.HasNext() {
+			event, err := iter.Next()
+			require.NoError(t, err)
+			if event.GetEventType() == enums.EVENT_TYPE_NEXUS_OPERATION_STARTED {
+				targetEvent = event
+				break
+			}
+		}
+		require.NotNil(t, targetEvent)
+		require.Len(t, targetEvent.GetLinks(), 1)
+		link := targetEvent.GetLinks()[0]
+		require.Equal(t, tc.testConfig.Namespace, link.GetWorkflowEvent().GetNamespace())
+		require.Equal(t, handlerWfID, link.GetWorkflowEvent().GetWorkflowId())
+		require.NotEmpty(t, link.GetWorkflowEvent().GetRunId())
+		requireProtoEqual(
+			t,
+			&common.Link_WorkflowEvent_EventReference{
+				EventType: enums.EVENT_TYPE_WORKFLOW_EXECUTION_STARTED,
+			},
+			link.GetWorkflowEvent().GetEventRef())
+		handlerRunID := link.GetWorkflowEvent().GetRunId()
```

Comment:
nit: you could have just used `requireProtoEqual` on the entire struct.

Reply:
I cannot because I don't have the handler run id.